### PR TITLE
Sb 162 add porduct first step

### PR DESCRIPTION
**Pull request by djanbolotov**
Changes:
Refactor wrong structure (components, inline styles)
Added first step for add product

Task: [Jira](https://team-jfqe9wlknaql.atlassian.net/jira/software/c/projects/SB/boards/8?assignee=5fc8d60bf8b0120069fd3f2e&selectedIssue=SB-162)

### **General**
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet
### Frontend
- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

Снимок экрана 2024-07-08 в 13 42 28
Снимок экрана 2024-07-08 в 13 42 47

### CODE REVIEW

File: src/pages/add-product/components/field-description/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import { StyledDescriptionBox, TitleTypography, DescriptionTypography } from "./styles.ts"
+
+interface Props {
```
Comment (VladManukian) [inline src/pages/add-product/components/field-description/index.tsx]:
```suggestion
type Props = {
```

File: src/pages/add-product/components/field-description/styles.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const TitleTypography = styled(Typography)`
+  color: ${theme.palette.primary.main};
+  margin-bottom: 8px;
+  font-size: ${theme.typography.playfairDisplayBold};
+`;
+
+const DescriptionTypography = styled(Typography)`
+  color: ${theme.palette.fontGray};
+  font-size: ${theme.typography.dmSans};
+`;
```
Comment (VladManukian) [inline src/pages/add-product/components/field-description/styles.ts]: For all styles with theme.
```suggestion
const DescriptionTypography = styled(Typography)(({ theme }) => ({
  color: theme.palette.fontGray,
  fontSize: theme.typography.dmSans,
}));
```
or 
```suggestion
const DescriptionTypography = styled(Typography)(({ theme }) => ({
  color: theme.palette.fontGray,
  ...theme.typography.dmSans,
}));
```

File: src/pages/add-product/components/image-list/index.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    setSelectedImage: (id: string) => void;
+    handleReplaceImage: (event: React.ChangeEvent<HTMLInputElement>, id: string) => void;
+    removeImage: (id: string) => void;
+}
+
+const ImageList: React.FC<Props> = ({ images, setPrimaryImage, setSelectedImage, handleReplaceImage, removeImage }) => {
+    return (
+        <StyledImageContainer>
+            {images.map((image) => (
+                <ImageBox key={image.id} selected={image.selected} onClick={() => setSelectedImage(image.id)}>
```
Comment (VladManukian) [inline src/pages/add-product/components/image-list/index.tsx]: Linter is not runned. Check it.

File: src/pages/add-product/components/image-list/styles.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import styled from '@emotion/styled';
```
Comment (VladManukian) [inline src/pages/add-product/components/image-list/styles.ts]: It is preferable to use 
```suggestion
import { styled } from "@mui/material";
```
for style

File: src/pages/add-product/components/image-uploader/index.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                        type="file"
+                        accept=".png, .jpg, .jpeg"
+                        onChange={handleImageUpload}
+                        multiple
+                    />
+                    <Button variant="contained" component="span">
+                        {t("AddVendorProduct.addImage")}
+                    </Button>
+                </label>
+                <Typography variant="body2" color="textSecondary">
```
Comment (VladManukian) [inline src/pages/add-product/components/image-uploader/index.tsx]: Move to custom style an use palette for color.

File: src/pages/add-product/components/image-uploader/index.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                    {t("AddVendorProduct.orDropImageToUpload")}
+                </Typography>
+                {error && (
+                    <div style={{ color: colors.red }}>{error}</div>
```
Comment (VladManukian) [inline src/pages/add-product/components/image-uploader/index.tsx]: 1.3.2.
1.3.4.

File: src/pages/add-product/components/image-uploader/styles.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import styled from "@emotion/styled";
+import { Box } from "@mui/material";
+
+const UploadContainer = styled(Box)({
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    textAlign: 'center',
+    alignItems: 'center',
+    border: '2px dashed #cccccc',
```
Comment (VladManukian) [inline src/pages/add-product/components/image-uploader/styles.ts]: 1.3,2

File: src/pages/add-product/hooks/useImageUpload.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+                    primary: false,
+                    selected: false,
+                };
+            });
+            setImages((prevImages) => prevImages.concat(fileArray));
+        }
+    };
+
+    const handleReplaceImage = (event: React.ChangeEvent<HTMLInputElement>, id: string) => {
+        if (event.target.files && event.target.files[0]) {
```
Comment (VladManukian) [inline src/pages/add-product/hooks/useImageUpload.ts]: Start linter, may be magic number

**Review by VladManukian — COMMENTED** [review]
Use {styled } from "@mui/material";
Linter is not running.
Check you pre-commit, may be you need another path.
Use Type instead of interface

File: src/components/profile-menu/styles.ts [hunk 9]
```diff
@@ -1,7 +1,7 @@
 import { Box, ButtonProps, styled } from "@mui/material";
 
-import { BaseButton } from "~/components/index.ts";
-import { colors, fontSizes } from "~/libs/constants/index.ts";
+import { BaseButton } from "~/components";
+import { colors, fontSizes } from "~/libs/constants";
```
Comment (likeri29) [inline src/components/profile-menu/styles.ts:5]: You do not have file extensions here. Run linter

File: src/components/tabs-board/styles.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
+import { styled } from "@mui/material/styles";
+import {Box, BoxProps, Typography} from "@mui/material";
+
+interface StyledNumberCircleProps extends BoxProps {
+	isBeforeActive?: boolean;
+}
+
```
Comment (likeri29) [inline src/components/tabs-board/styles.ts:7]: We do not use interfaces. Change to type
```suggestion
type StyledNumberCircleProps = {isBeforeActive?: boolean} & BoxProps
```

File: src/components/tabs-board/styles.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+		display: { tablet: "flex", xs: "none" },
+	},
+	width: "3em",
+}));
+
+export const StyledLabelText = styled(Typography)(({ theme }) => ({
+	color: theme.palette.primary.main,
+	fontSize: theme.typography.dmSansBold.fontSize,
+	fontWeight: "bold",
+}));
```
Comment (likeri29) [inline src/components/tabs-board/styles.ts:39]: all exports at the bottom

File: src/pages/add-product/components/first-step/index.tsx [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+import { useTranslation } from "react-i18next";
+
+import { FieldDescription } from "../field-description/index.tsx";
+import { ImageList } from "../image-list";
+import { ImageUploader } from "../image-uploader/index.tsx";
+import { StyledBox, StyledFormContainer } from "./styles.ts";
+import { useImageUpload } from "../../hooks/useImageUpload.ts";
+import { imagesValidation } from "~/pages/add-product/validation/add-images";
+import { FormButtons } from "~/pages/profile-board/components/buttons";
+
```
Comment (likeri29) [inline src/pages/add-product/components/first-step/index.tsx:15]: again file extensions

**Comment by likeri29** [comment]
Also, I have a question, why did you move the profile menu component from the profile folder to components? This component is used only on the profile page

File: src/components/tabs-board/styles.ts (hunk 11 above)
Comment (likeri29) [reply src/components/tabs-board/styles.ts:39]: and for fontWeight use theme

**Comment by lyudad** [comment]
Screenshot 2024-07-08 at 23 27 39

**Comment by djanbolotov** [comment]
> Screenshot 2024-07-08 at 23 27 39
fixed
Снимок экрана 2024-07-09 в 10 56 02

**Pull request merged**